src/ophyd_async/core/signal.py

```diff
@@ -272,6 +285,119 @@ def soft_signal_r_and_backend(
     return (signal, backend)
 
 
+async def _verify_readings(
+    func: Callable[[], Mapping[str, Reading] | Awaitable[Mapping[str, Reading]]],
+    expectation: Mapping[str, Reading],
+) -> None:
+    """Take a read/read_configuration function that return a dictionary and
+        compare it with the expected result (expectation)
+
+    Parameters
+    ----------
+    func:
+        read/read_configuration function.
+
+    expectation:
+        The expected value from the readable/Configurable.
+
+    Notes
+    -----
+    Example usage::
+        await _verify_readings(readable.read, reading)
+    Or::
+        await _verify_readings(configurable.read_configuration, configuration)
+
+    """
+    result = await func()
+
+    assert result == expectation
+
+
+async def assert_value(signal: SignalR[T], value: Any) -> None:
+    """Assert a signal's value and compare it an expected signal.
+
+    Parameters
+    ----------
+    signal:
+        signal with get_value.
+    value:
+        The expected value from the signal.
+
+    Notes
+    -----
+    Example usage::
+        await assert_value(signal, value)
+
+    """
+    assert await signal.get_value() == value
+
+
+async def assert_reading(
+    readable: AsyncReadable, reading: Mapping[str, Reading]
+) -> None:
+    """Assert readings from readable.
+
+    Parameters
+    ----------
+    readable:
+        Callable with readable.read function that generate readings.
+
+    reading:
+        The expected readings from the readable.
+
+    Notes
+    -----
+    Example usage::
+        await assert_reading(readable, reading)
+
+    """
+    await _verify_readings(readable.read, reading)
+
+
+async def assert_configuration(
+    configurable: AsyncConfigurable,
+    configuration: Mapping[str, Reading],
+) -> None:
+    """Assert readings from Configurable.
+
+    Parameters
+    ----------
+    configurable:
+        Configurable with Configurable.read function that generate readings.
+
+    configuration:
+        The expected readings from configurable.
+
+    Notes
+    -----
+    Example usage::
+        await assert_configuration(configurable configuration)
+
+    """
+    await _verify_readings(configurable.read_configuration, configuration)
```

Comment:
Please can we now ditch `_verify_readings` and just write:
```suggestion
    assert await configurable.read_configuration() == configuration
```

Reply:
done, so simple, only took 23 attempt...